lib/crystalball/graphql/map_generator/graphql_strategy.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+require 'crystalball/map_generator/base_strategy'
+
+module Crystalball
+  module GraphQL
+    module MapGenerator
+      # Map generator strategy to build map of GraphQL type definitions
+      # that might affect an example.
+      # Patches `GraphQL::ObjectType.define` to get the path of files that call it.
+      class GraphQLStrategy
+        include ::Crystalball::MapGenerator::BaseStrategy
+        include ::Crystalball::MapGenerator::Helpers::PathFilter
+
+        class << self
+          attr_accessor :current_schema, :type_definition_paths, :patch_applied
+
+          # Adds the type definition paths of each of the `other_keys`
+          # to the type definition paths of `key`
+          def augment_type_definition_paths(key, other_keys)
+            type_definition_paths[key.object_id] |= other_keys
+                                                    .map { |t| type_definition_paths[t.object_id] }
+                                                    .flatten
+                                                    .compact
+          end
+
+          # Checks whether `::GraphQL::Schema` is defined and, if it is, applies
+          # the patch to it. It needs to be this way because `graphql` gem is weird.
+          def apply_patch
```

Comment:
why do you need this method at all? Why would you want to include new strategy which is senseless if you don't have GraphQL?

Reply:
GraphQL specs break when we require it in this file. I'll just keep the bang version to be called during `#call` where we're sure `graphql` has been loaded already.